teamB/eungSooKim/services/userService.js

```diff
@@ -0,0 +1,33 @@
+const userDao = require("../models/userDao");
+
+const signUp = async (name, email, password, profileImage) => {
+  // password validation using REGEX
+  const pwValidation = new RegExp(
+    "^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,20})"
+  );
+  if (!pwValidation.test(password)) {
+    const err = new Error(
+      "비밀번호는 영어,숫자,특수문자를 포함해 8-20자로 만들어주세요"
+    );
+    err.statusCode = 409;
+    throw err;
```

Comment:
1. 에러 메시지는 영어로 작성해 주시고, 디테일한 내용 보다는 간결하게 어떤 에러인지 파악할 수 있으면 됩니다. 해당 메시지는 사용자가 보는 것이 아닌, 프론트엔드에서 받아가는 메시지 입니다.
2. 상태코드 409는 PUT method 사용시 데이터 충돌 날 때 주로 쓰는 코드 입니다.
```suggestion
  if (!pwValidation.test(password)) {
    const err = new Error(
      "INVALID PASSWORD"
    );
    err.statusCode = 400;
    throw err;
```